### ISSUE TITLE
test_shim tests no longer completely pass because of changes in python circa v3.00.48

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarrac#138]
all versions of the sarrac/metpx-sr3c fail to run the complete test_shim suite since some change was made to the python implementation.  Sample failure:

```


2023-12-19 10:23:14,143 [INFO] published: { "pubTime":"20231219152314.14052573", "baseUrl":"file:/home/peter/Sarracenia/metpx-sr3c/shim_dirA", "relPath":"dirthree/filefive", "topic":"v03.post.dirthree", "toto":"pig", "fileOp" : { "remove":""}}

RESULT: Good! 1 remove posts from 165 symlink in a non sub-dir. as expected
RESULT: comment 160 shim copy posting end
RESULT: BAD tree differences in 0 files, and 4 links

posts: {'sha512': 12, 'link': 6, 'rename': 7, 'remove': 2, 'directory': 2}
RESULT: summary: good: 29, bad 1, total 28
make: *** [Makefile:137: test_shim_copy_mirror] Error 1

```

The normal shim tests used to all pass, 

```

fractal% tail Makefile

test_shim_copy_flatten:
	-./shim_copy_flatten.sh >shim_copy_flatten.log 2>&1
	python3 ./check_shim_test.py shim_copy_flatten.log 

test_shim_copy_baseDir:
	-./shim_copy_baseDir.sh >shim_copy_baseDir.log 2>&1
	python3 ./check_shim_test.py shim_copy_baseDir.log 

test_shim: test_shim_post test_shim_copy_strip test_shim_copy_mirror test_shim_copy_mirror_sftp test_shim_copy_baseDir
fractal% 

```

and we can see it runs 1 test to confirm posting is ok, then four different groups of settings to do mirroring with: 

* copy_strip -- uses a strip count to remove directories to create destination paths.
* copy_mirror -- post_baseUrl (to derive baseDir) combined with a directory to create destination paths.... using file: urls.
* copy_mirror_sftp -- as above, but using sftp URLs instead of file.
* copy_mirror_baseDir -- 

These formerly passed with tags 3.23.11p1, and p2.  noticed problem when preparing p3, and went back and the older versions don't pass anymore.

**Comment by petersilva** [comment]
result of git bisection:

```
6c0535c3ae24cbd73de305849422520b9ae60e38 is the first bad commit
commit 6c0535c3ae24cbd73de305849422520b9ae60e38
Author: Peter Silva <peter@bsqt.homeip.net>
Date:   Fri Nov 24 13:49:06 2023 -0500

     fixing one problem related to #639 is relative paths being treated as absolute when baseUrl.path=''

 sarracenia/flow/__init__.py | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)
fractal% git show 6c0535c3ae24cbd73de305849422520b9ae60e38
commit 6c0535c3ae24cbd73de305849422520b9ae60e38 (HEAD, refs/bisect/bad)
Author: Peter Silva <peter@bsqt.homeip.net>
Date:   Fri Nov 24 13:49:06 2023 -0500

     fixing one problem related to #639 is relative paths being treated as absolute when baseUrl.path=''

diff --git a/sarracenia/flow/__init__.py b/sarracenia/flow/__init__.py
index 4cd38cdb..501e6130 100644
--- a/sarracenia/flow/__init__.py
+++ b/sarracenia/flow/__init__.py
@@ -1694,17 +1694,13 @@ class Flow:
             u = sarracenia.baseUrlParse(msg['baseUrl']) 
             logger.debug( f"baseUrl.path= {u.path} ")
             if remote_path:
-                if u.path != '/' and remote_path:
-                    if remote_path[0] == '/':
-                        remote_path = u.path + remote_path
-                    else:
+                if u.path: 
+                    if ( u.path[-1] != '/' ) and ( remote_path[0] != '/' ) :
                         remote_path = u.path + '/' + remote_path
+                    else:
+                        remote_path = u.path + remote_path
 
-                # relPath does not contain a prefix / , add it for cdir
-                if remote_path[0] != '/':
-                    cdir = '/' + remote_path
-                else:
-                    cdir = remote_path
+                cdir = remote_path
             else:
                 if u.path:
                     cdir=u.path
fractal% 

```

**Comment by petersilva** [comment]
ok, so adding a slash to the end of post_baseDir makes the test pass.  When I retest p2 with the same patch it also passes.

between p2 and p3 the same patch was made 0c7839ef9eef66e153c1276f3a1fe78dc4fd817c so there is no need
to patch so far.

I advance to the tip of the development trees in both implementations and results
remain good.

**Comment by petersilva** [comment]
now I see that the tests that are used in the github Action are not those in make test_shim, but a specific, other set of tests, that include test_shim_copy_strip_slash.
and that still fails.

**Comment by petersilva** [comment]
this is the bad patch for strip_slash:

```
fractal% git bisect good
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[6c0535c3ae24cbd73de305849422520b9ae60e38]  fixing one problem related to #639 is relative paths being treated as absolute when baseUrl.path=''
fractal% git bisect bad
6c0535c3ae24cbd73de305849422520b9ae60e38 is the first bad commit
commit 6c0535c3ae24cbd73de305849422520b9ae60e38
Author: Peter Silva <peter@bsqt.homeip.net>
Date:   Fri Nov 24 13:49:06 2023 -0500

     fixing one problem related to #639 is relative paths being treated as absolute when baseUrl.path=''

 sarracenia/flow/__init__.py | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)
fractal% 

```

**Comment by petersilva** [comment]
It's the same patch, and the same fix (but in another test.)

**Comment by petersilva** [comment]
for copy_slash... 8b80ef77689a75b4399f015e607c9db5ed4230ed

**Comment by petersilva** [comment]
it's fixed... released in 3.00.54 at worst...
